tests/unit/cluster/cli.tcl

```diff
@@ -54,8 +54,10 @@ start_multiple_servers 3 [list overrides $base_conf] {
     }
 
     test "Verify command got unblocked after resharding" {
-        # this (read) will wait for the node3 to realize the new topology
-        assert_error {*MOVED*} {$node3_rd read}
+        # this (read) will wait for the node3 to realize the new topology and then unblock via MOVED,
+        # or it will unblock with CLUSTERDOWN, depending on the cluster state when clientsCron is called.
+        catch {$node3_rd read} err
+        assert {([string range $err 0 4] eq {MOVED}) || ([string range $err 0 10] eq {CLUSTERDOWN})}
```

Comment:
i think it's better to use `string match` on a pattern rather than `string range`

Reply:
yes, updated